lib/pengine/utils.c

```diff
@@ -2807,3 +2792,26 @@ pe__rsc_running_on_any_node_in_list(pe_resource_t *rsc, GListPtr node_list)
 
     return false;
 }
+
+bool
+pe__rsc_allowed_on_any_node_in_list(pe_resource_t *rsc, GListPtr node_list)
+{
+    GHashTableIter iter;
+    pe_node_t *node = NULL;
+
+    g_hash_table_iter_init(&iter, rsc->allowed_nodes);
+    while (g_hash_table_iter_next(&iter, NULL, (void **)&node)) {
+        if (pcmk__str_in_list(node_list, node->details->uname)) {
+            return true;
+        }
+    }
+
+    g_hash_table_iter_init(&iter, rsc->known_on);
+    while (g_hash_table_iter_next(&iter, NULL, (void **)&node)) {
+        if (pcmk__str_in_list(node_list, node->details->uname)) {
+            return true;
+        }
+    }
```

Comment:
This behaves slightly differently from what you had before. Before it would only check known_on if allowed_nodes were empty, now it checks known_on if allowed_nodes doesn't contain a target node.

Reply:
Done